allennlp/models/semantic_role_labeller.py

```diff
@@ -0,0 +1,191 @@
+        verb_indicator: ``IndexField``, required.
+            The index of the verb whose arguments we are labelling.
+
+        Returns
+        -------
+        A Dict containing:
+
+        tags : List[str]
+            A list the length of the text input, containing the predicted (argmax) tag
+            from the model per token.
+        class_probabilities : numpy.Array
+            An array of shape (text_input_length, num_classes), where each row is a
+            distribution over classes for a given token in the sentence.
+        """
+        instance = Instance({"tokens": text_field, "verb_indicator": verb_indicator})
+        instance.index_fields(self.vocab)
+        model_input = instance.as_array(instance.get_padding_lengths())
+        torch_input = arrays_to_variables(model_input)
+        # TODO(Mark): Make the data API always return tensors with batch dimensions at every abstraction level.
+        # Add a batch dimension by unsqueezing, because pytorch doesn't support inputs without one.
+        torch_input["tokens"]["tokens"].data.unsqueeze_(0)
+        torch_input["verb_indicator"].data.unsqueeze_(0)
+
+        output_dict = self.forward(**torch_input)
+
+        # Remove batch dimension, as we only had one input.
+        predictions = output_dict["class_probabilities"].data.squeeze(0)
+        transition_matrix = self.get_viterbi_pairwise_potentials()
+
+        max_likelihood_sequence, _ = viterbi_decode(predictions, transition_matrix)
+        tags = [self.vocab.get_token_from_index(x, namespace="tags")
+                for x in max_likelihood_sequence]
+
+        return {"tags": tags, "class_probabilities": predictions.numpy()}
+
+    def get_viterbi_pairwise_potentials(self):
+        """
+        Generate a matrix of pairwise transition potentials for the BIO labels.
+        The only constraint implemented here is that I-XXX tags must be preceded
+        by either an identical I-XXX tag or a B-XXX tag (i.e they have a pairwise
+        potential of -inf).
+
+        Returns
+        -------
+        transition_matrix : torch.Tensor
+            A (num_tags, num_tags) matrix of pairwise potentials.
+        """
+        all_tags = self.vocab.get_index_to_token_vocabulary("tags")
+        num_tags = len(all_tags)
+        transition_matrix = torch.zeros([num_tags, num_tags])
+
+        for i, previous_tag in all_tags.items():
+            for j, tag in all_tags.items():
+                # I tags can only be preceded by themselves or
+                # their corresponding B tag.
+                if i != j and tag[0] == 'I' and not previous_tag == 'B' + tag[1:]:
+                    transition_matrix[i, j] = float("-inf")
+        return transition_matrix
+
+    @classmethod
+    def from_params(cls, vocab: Vocabulary, params: Params) -> 'SemanticRoleLabeller':
+        text_field_embedder = TextFieldEmbedder.from_params(vocab, params.pop("text_field_embedder"))
```

Comment:
I'd recommend putting in defaults here that replicate Luheng's model, like I did for the BiDAF code.